### PR TITLE
Refactoring

### PR DESCRIPTION
**Pull request by AlekseyKorshuk**
* Code refactored with overriding needed methods of each pipeline for ONNX models

* Imports updated

### CODE REVIEW

File: optimum_transformers/generation_utils.py [hunk 1]
```diff
@@ -381,10 +381,10 @@ class GenerationMixin:
     """
     A class containing all of the functions supporting generation, to be used as a mixin in [`PreTrainedModel`].
     """
-    def __init__(self, model, onnx_model, use_onnx):
+    def __init__(self, model, onnx_model):
```
Comment (chainyo) [inline optimum_transformers/generation_utils.py:384]: We could add type hints to make it easier to debug and understand later.

File: optimum_transformers/pipelines/fill_mask.py [hunk 2]
```diff
@@ -1,222 +1,20 @@
...
-if is_tf_available():
-    import tensorflow as tf
-
-if is_torch_available():
-    import torch
+from .base import _forward_onnx, _warmup_onnx_graph
+from transformers import FillMaskPipeline
+import torch
 
 logger = logging.get_logger(__name__)
```
Comment (chainyo) [inline optimum_transformers/pipelines/fill_mask.py]: I saw some class with the logger and some others without it. Is it a legacy from HF Transformers package or is there a logger in this project ?

Comment (AlekseyKorshuk) [reply optimum_transformers/pipelines/fill_mask.py]: Exists in this project
Will update and rerequest review

**Review by chainyo — APPROVED** [review]
It looks really good! I like the way you used the Output class for each task. 

I was wondering if it will allow every models to be able to use onnx ? I mean sometimes there are classes like `OutputLM` or `OutputWithPast`...

**Comment by AlekseyKorshuk** [comment]
Models of specific type have the same output class. So such usage is ok

**Pull request merged**